packages/yarn-plugin-allow-scripts/test/index.spec.mts

```diff
@@ -0,0 +1,238 @@
+        },
+        undefined,
+        2
+      )}`
+    )
+    return {
+      error,
+      output: [],
+      exitCode: null,
+      signal: null,
+      out: '',
+      err: '',
+    };
+  }
+}
+
+test('install - errors when adding dependency with unspecified preinstall script', async (t: any) => {
+  const projectName = 'uninitialized';
+  const projectRoot = join(dirname(import.meta.url.replace(/^file:/, '')), 'projects', projectName);
+
+  try {
+    // delete any leftover test artifacts
+    rmSync(join(projectRoot, PACKAGE_JSON), { force: true })
+
+    // init project
+    const initRes = run(t, ['init', '-y'], projectRoot)
+    t.is(initRes.exitCode, 0);
+
+    // install the plugin
+    const importRes = run(t, ['plugin', 'import', normalize('../../../bundles/@yarnpkg/plugin-allow-scripts.js')], projectRoot)
+    t.is(importRes.err, '');
+    t.regex(importRes.out, /YN0000: Saving the new plugin in .yarn\/plugins\/@yarnpkg\/plugin-allow-scripts.cjs/);
+    t.is(importRes.exitCode, 0);
+
+    const addRes = run(t, ['add', '@lavamoat/preinstall-always-fail'], projectRoot)
+    t.regex(addRes.err, /allow-scripts detected attempted execution of unconfigured package script. \["@lavamoat\/preinstall-always-fail","preinstall"/);
+    t.is(addRes.exitCode, 1);
+
+    // trigger the plugin
+    const installRes = run(t, ['install'], projectRoot)
+    t.regex(installRes.out, /packages missing configuration:\n- @lavamoat\/preinstall-always-fail/);
+    t.regex(installRes.err, /@lavamoat\/allow-scripts has detected dependencies without configuration. explicit configuration required./);
+    t.is(installRes.exitCode, 1);
+  } finally {
+    cleanup(projectName);
+  }
+})
+
+test('install - allows adding and installing package with specified preinstall script', async (t: any) => {
+  const projectName = 'allowed';
+  const projectRoot = join(dirname(import.meta.url.replace(/^file:/, '')), 'projects', projectName)
+
+  try {
+    run(t, ['plugin', 'import', normalize('../../../bundles/@yarnpkg/plugin-allow-scripts.js')], projectRoot)
+    const addRes = run(t, ['add', '@lavamoat/preinstall-always-fail'], projectRoot)
+    t.is(addRes.exitCode, 0);
+
+    // trigger the plugin
+    const installRes = run(t, ['install'], projectRoot)
+    t.is(installRes.exitCode, 0);
+  } finally {
+    cleanup(projectName);
+  }
+})
+
+test('install - allows execution of allowed yarn classic dependencies', async (t: any) => {
+  const projectName = 'protected-configured';
+  const projectRoot = join(dirname(import.meta.url.replace(/^file:/, '')), 'projects', projectName)
+
+  try {
+    run(t, ['plugin', 'import', normalize('../../../bundles/@yarnpkg/plugin-allow-scripts.js')], projectRoot)
+
+    // trigger the plugin
+    const installRes = run(t, ['install', '--inline-builds'], projectRoot, {YARN_IGNORE_SCRIPTS: 'false'})
+    t.is(installRes.exitCode, 0);
+    t.true(existsSync(portablePath(join(projectRoot, 'node_modules', 'allowed-dep', 'foo'))));
```

Comment:
(should) I don't see anything specifically removing the file, so I'm assuming git cleanup does it, but it'd make the test more robust if the same assertion with `t.false` was executed before running `install`

Reply:
Yes, it's taken care of by `git clean -fdx`.

And if that would have not been the case, in any case that would already be caught in CI due to the dirty-git-state check? If anything is left over form cleanup, it will fail.